ui/src/app/icons/edit/component.tsx

```diff
@@ -0,0 +1,83 @@
+import {AfterViewInit, Component, ElementRef, ViewChild} from '@angular/core';
+
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+interface EditComponentState { style: object; }
+
+const style = {
+  height: 14,
+  width: 14,
+};
+
+const fillStyle = {
+  fill: '#4D4CA5'
+};
+
+class EditComponentReact extends React.Component<{}, EditComponentState> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+        style: style
```

Comment:
nit: generally we do two-space block indents (we should actually encode this somewhere in our linter..)

Reply:
Oof ... my IDE is weirdly messed up. The default indent is set 2 spaces but it still messes up sometimes. Will fix.

Reply:
Done.